docs/development/extensions-contrib/rocketmq-ingestion.md

```diff
@@ -0,0 +1,414 @@
+|`completionTimeout`|ISO8601 Period|The length of time to wait before declaring a publishing task as failed and terminating it. If this is set too low, your tasks may never publish. The publishing clock for a task begins roughly after `taskDuration` elapses.|no (default == `PT30M`)|
+|`lateMessageRejectionStartDateTime`|ISO8601 DateTime|Configure tasks to reject messages with timestamps earlier than this date time; for example if this is set to `2016-01-01T11:00Z` and the supervisor creates a task at *`2016-01-01T12:00Z`*, messages with timestamps earlier than *`2016-01-01T11:00Z`* will be dropped. This may help prevent concurrency issues if your data stream has late messages and you have multiple pipelines that need to operate on the same segments (e.g. a realtime and a nightly batch ingestion pipeline).|no (default == none)|
+|`lateMessageRejectionPeriod`|ISO8601 Period|Configure tasks to reject messages with timestamps earlier than this period before the task was created; for example if this is set to `PT1H` and the supervisor creates a task at *`2016-01-01T12:00Z`*, messages with timestamps earlier than *`2016-01-01T11:00Z`* will be dropped. This may help prevent concurrency issues if your data stream has late messages and you have multiple pipelines that need to operate on the same segments (e.g. a realtime and a nightly batch ingestion pipeline). Please note that only one of `lateMessageRejectionPeriod` or `lateMessageRejectionStartDateTime` can be specified.|no (default == none)|
+|`earlyMessageRejectionPeriod`|ISO8601 Period|Configure tasks to reject messages with timestamps later than this period after the task reached its `taskDuration`; for example if this is set to `PT1H`, the `taskDuration` is set to `PT1H` and the supervisor creates a task at *`2016-01-01T12:00Z`*, messages with timestamps later than *`2016-01-01T14:00Z`* will be dropped. **Note:** Tasks sometimes run past their task duration, for example, in cases of supervisor failover. Setting `earlyMessageRejectionPeriod` too low may cause messages to be dropped unexpectedly whenever a task runs past its originally configured task duration.|no (default == none)|
+
+#### More on consumerProperties
+
+Now `consumerProperties` only support one property. This must contain a property `nameserver.url` with a list of `RocketMQ` name servers in the form: `<NAMESREV_1>:<PORT_1>;<NAMESREV_2>:<PORT_2>;...`.
+
+#### Specifying data format
+
+`RocketMQ` indexing service supports both [`inputFormat`](../../ingestion/data-formats.md#input-format) and [`parser`](../../ingestion/data-formats.md#parser) to specify the data format.
+The `inputFormat` is a new and recommended way to specify the data format for `RocketMQ` indexing service,
+but unfortunately, it doesn't support all data formats supported by the legacy `parser`.
```

Comment:
is `parser` supported by this extension ?

Reply:
Yes. Actually, parser is supported by SeekableStreamIndexTaskRunner. It's a deprecated interface and will be replaced by InputFormat.